app/presenters/user_analytics_presenter.rb

```diff
@@ -16,22 +16,32 @@ def initialize(current_facility)
     @current_facility = current_facility
     @daily_period_list = period_list_as_dates(:day, DAYS_AGO)
     @monthly_period_list = period_list_as_dates(:month, MONTHS_AGO)
-    @user_analytics = UserAnalyticsQuery.new(current_facility, days_ago: DAYS_AGO, months_ago: MONTHS_AGO)
+
+    @last_refreshed_at =
+      Rails.cache.fetch(Rails.application.config.app_constants[:MATVIEW_REFRESH_TIME_KEY]).presence || Time.current
+
+    @user_analytics = UserAnalyticsQuery.new(current_facility,
+                                             days_ago: DAYS_AGO,
+                                             months_ago: MONTHS_AGO,
+                                             fetch_until: @last_refreshed_at)
   end
 
   def statistics
-    @statistics ||= {
-      daily: daily_stats,
-      monthly: monthly_stats,
-      all_time: all_time_stats,
-      trophies: trophy_stats,
-      metadata: {
-        is_diabetes_enabled: false,
-        last_updated_at: I18n.l(Time.current),
-        formatted_next_date: display_date(Time.current + 1.day),
-        today_string: I18n.t(:today_str)
-      }
-    }
+    @statistics ||=
+      Rails.cache.fetch(statistics_cache_key) do
```

Comment:
If I understand correctly, we are introducing caching here because we now can, since the data is up to the last matview refresh, and not up to the present time when the page is fetched.

Reply:
Yeah, just stuck in caching as a natural extension to capping data until the last refresh.